### PR TITLE
Greg/save submissions

### PR DESCRIPTION
**Pull request by GodGreg**
This took about 12 hours lmao. I left the programs update commit out because it was buggy and tired (will do tomorrow). So I'll leave you with 1000+ lines

Submission API
All question types updated to support submissions
Submission is saved every time something changes focus like in formCreation. Final save on submit adds the submission date and checks if the form closed during their submission
ThemeColour implemented on submissions

https://www.notion.so/uwblueprintexecs/API-endpoint-for-saving-form-responses-and-form-submission-6a5df7e47e364a5893470b9a53f1fe18
https://www.notion.so/uwblueprintexecs/Form-closed-page-3f37175ee43847998ef22821d969411c

Preview form saves submissions for testing purposes, we can disable once we release to prod.

### CODE REVIEW

File: src/Components/FormCreation/CardComponents/FileQuestion.tsx [hunk 1]
```diff
@@ -16,37 +18,55 @@ type Props = {
...
-    //returns the url to the file in aws
-    await fileUpload("arbitrium", file.name, formData);
-    //TODO Save to DB
+    if (file) {
+      //todo: check if the file extension is valid
+      const formData = new FormData();
+      formData.append("file", file); // appending file
+
+      //returns the url to the file in aws
+      const result = await fileUpload("arbitrium", file.name, formData);
```
Comment (ahmedhamodi) [inline src/Components/FormCreation/CardComponents/FileQuestion.tsx:65]: Might be a dumb question but in a previous file I saw "arbitrium-public". I assume conversely we have "arbitrium-private"? Should this line be using one of those two?

File: src/Components/FormCreation/CardComponents/FileQuestion.tsx [hunk 2]
```diff
@@ -60,68 +80,92 @@ function FileQuestion({
...
+  // //Create a link to download the file :)
+  // if (!loadFile.isPending && loadFile.value && loadFile.value.Body) {
+  //   const bytes = new Uint8Array(loadFile.value.Body.data); // pass your byte response to this constructor
+  //   const blob = new Blob([bytes], { type: "application/octet-stream" }); // change resultByte to bytes
+  //   link = window.URL.createObjectURL(blob);
+
+  //   // const headerImg = document?.querySelector<HTMLImageElement>("#image");
+  //   // // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  //   // headerImg!.src = link;
+  // }
```
Comment (ahmedhamodi) [inline src/Components/FormCreation/CardComponents/FileQuestion.tsx:94]: Is there a future use case for this or should we delete it?

File: src/Components/FormCreation/CardComponents/SelectQuestion.jsx [hunk 3]
```diff
@@ -138,88 +152,191 @@ function SelectQuestion({
...
+  };
+
+  const handleMultipleChoiceSelect = (event) => {
+    setSelected({ ...selectedInit, [event.target.name]: event.target.checked });
+    onChange({ ...selectedInit, [event.target.name]: event.target.checked });
+  };
+
+  // useEffect(() => {
+  //   onChange(selected);
+  // }, [selected, onChange]);
```
Comment (ahmedhamodi) [inline src/Components/FormCreation/CardComponents/SelectQuestion.jsx:171]: Is this needed?

File: src/Components/FormCreation/CreateEditForm.js [hunk 4]
```diff
@@ -81,12 +82,17 @@ function CreateEditForm() {
   const [showFormSettings, setShowFormSettings] = useState(false);
   const [showMoveSectionsDialog, setShowMoveSectionsDialog] = useState(false);
   const [activeSection, setActiveSection] = useState(0);
-  const programId = appUser.currentProgram;
+  //const programId = appUser.currentProgram;
```
Comment (ahmedhamodi) [inline src/Components/FormCreation/CreateEditForm.js:85]: `programId` looks like it's being passed in as a prop now, so can probably remove this.

**Review by ahmedhamodi — COMMENTED** [review]
First pass comments. Testing locally now.

**Comment by ahmedhamodi** [comment]
Not sure if something broke but I attempted to upload a photo and I'm getting this spinner for a long time. Perhaps an issue with the bucketname (arbitrium vs arbitrium-public?)
image

**Comment by ahmedhamodi** [comment]
Form closed page looks alright, a few issues though:

- we've **saved** a copy...
- overflow on first line

See image:
image

**Comment by ahmedhamodi** [comment]
Also last thing I noticed, I don't know if my colour choice appeared on the submission side (or perhaps didn't save on the creation side). As you can see I picked the turquoise colour on the creation side but on the submission side I see the standard blue.

File: src/Components/FormCreation/CardComponents/FileQuestion.tsx (hunk 1 above)
Comment (GodGreg) [reply src/Components/FormCreation/CardComponents/FileQuestion.tsx:65]: "aribtrium" is the private one.

File: src/Components/FormCreation/CardComponents/FileQuestion.tsx (hunk 2 above)
Comment (GodGreg) [reply src/Components/FormCreation/CardComponents/FileQuestion.tsx:94]: Yes. When displaying on decision canvas we will need this code. Whether we purpose this component as a read-only version or move it somewhere else

File: src/Components/FormCreation/CardComponents/SelectQuestion.jsx (hunk 3 above)
Comment (GodGreg) [reply src/Components/FormCreation/CardComponents/SelectQuestion.jsx:171]: no, removed

File: src/Components/FormCreation/CreateEditForm.js (hunk 4 above)
Comment (GodGreg) [reply src/Components/FormCreation/CreateEditForm.js:85]: Good catch

**Comment by GodGreg** [comment]
> Also last thing I noticed, I don't know if my colour choice appeared on the submission side (or perhaps didn't save on the creation side). As you can see I picked the turquoise colour on the creation side but on the submission side I see the standard blue.

I could not reproduce the photo error, lets sync later to check that out.

The colours not saving was due to publishing a form overwriting the settings. I fixed this

AND, if the form is closed or gives an error message then the colours don't load as we don't even load the form contents

**Review by ahmedhamodi — APPROVED** [review]

**Pull request merged**